File: AddTwoNumbers.py
from typing import Optional
import functools as ft
from functools import partial as par
def F(*z):
    z = [*z]
    return [*ft.reduce(lambda x, y: map(y, x), [z[:1]] + z[1:])][0]
FF = lambda *z: [*ft.reduce(lambda x, y: map(y, x), z)]
fyx = lambda f, *x: lambda *y: f(*y, *x)
class ListNode:
    def __init__(self, val = 0, next = None):
        self.val = val
        self.next = next
class Solution:
    def toListNumber(x):
        out = []
        while (x):
            out.append(x.val)
            x = x.next
        return out
    def toListNode(x):
        out = ListNode()
        previous = tail = out
        for xx in x:
            tail.val = xx
            tail.next = ListNode() 
            previous = tail
            tail = tail.next
        previous.next = 0
        return out
    def output(x, fout):
        out = ''
        for xx in x:
            out += str(xx) + ','
        fout.write(out[:-1])
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]
                        ) -> Optional[ListNode]:
        base = 0
        head = tail = ListNode()
        while l1 or l2:
            num1 = num2 = 0
            if (l1):
                num1 = l1.val
            if (l2):
                num2 = l2.val
            if num1 + num2 + base < 10:
                tail.val = num1 + num2 + base
                base = 0
            else:
                tail.val = num1 + num2 + base - 10
                base = 1
            if (l1):
                l1 = l1.next
            if (l2):
                l2 = l2.next
            if l1 or l2:
                tail.next = ListNode()
                tail = tail.next
        if (base):
            tail.next = ListNode(1)
        return head

fin = open("oo.xx", "r");
fout = open("xx.oo", "w");

input1 = [int(x) - int('0') for x in fin.readline()[:-1].split(',')]
input2 = [int(x) - int('0') for x in fin.readline()[:-1].split(',')]
input1, input2 = FF([input1, input2], Solution.toListNode)

#F(Solution().addTwoNumbers(input1, input2), Solution.toListNumber, print)
F(Solution().addTwoNumbers(input1, input2),
    Solution.toListNumber, fyx(Solution.output, fout))

File: AddTwoNumbers.c
#include<stdio.h>
#include<string.h>
#include<stdlib.h>
#include<stdbool.h>
#include<math.h>
struct ListNode
{
    int val;
    struct ListNode *next;
};

void output(struct ListNode* l, char split, FILE *fout)
{
	struct ListNode *tail = l;
	while (tail)
	{
		if (tail->next)
			fprintf(fout, "%d%c", tail->val, split);
		else
			fprintf(fout, "%d", tail->val);
		tail = tail->next;
	}
	fputc('\n', fout);
}

struct ListNode* addTwoNumbers(struct ListNode* l1, struct ListNode* l2)
{
	int base = 0;
    	bool next1 = 1, next2 = 1;
    	struct ListNode *head = malloc(sizeof(struct ListNode)), *tail = head;
    	do 
    	{
			int num1 = 0, num2 = 0;
			if (l1) num1 = l1->val;
			if (l2) num2 = l2->val;
			if (l1 && !l1->next)    next1 = 0;
			if (l2 && !l2->next)    next2 = 0;
			if (num1 + num2 + base < 10)
			{
				tail->val = num1 + num2 + base;
				base = 0;
			} else {
				tail->val = num1 + num2 + base - 10;
				base = 1;
			}
			if (next1 || next2)
			{
				tail->next = malloc(sizeof(struct ListNode));
				tail = tail->next;
			}
			if (l1)	l1 = l1->next;
			if (l2) l2 = l2->next;
			if (!next1 && !next2)	tail->next = 0;
    	} while (l1 || l2);
    	if (base)
    	{
			tail->next = malloc(sizeof(struct ListNode));
			tail->next->next = 0;
			tail->next->val = 1;
    	}
    	return head;
}

int main()
{
	FILE *fin = fopen("oo.xx", "r");
	FILE *fout = fopen("xx.oo", "w");
	struct ListNode *l1, *l2, *tail, *previous;
	l1 = malloc(sizeof(struct ListNode));
	l2 = malloc(sizeof(struct ListNode));
	tail = l1;
	char temp;
	bool flag = 0;
	while (fscanf(fin, "%c", &temp) != EOF)
	{
		if (temp == ' ' || temp == ',')	continue;
		if (temp == '\n')
		{
			free(tail);
			previous->next = 0;
			if (flag)	break;
			tail = l2;
			flag = 1;
			continue;
		}
		tail->val = temp - '0';
		tail->next = malloc(sizeof(struct ListNode));
		previous = tail;
		tail = tail->next;
	}
	output(addTwoNumbers(l1, l2), ',', fout);
	return 0;
}

